In [1]:
from math import factorial as fact
from scipy.special import comb

###### Question 12b:

Verify the following identity for $n = 1,2,3,4,5$

###### In Python

In [35]:
# combinatorial identities 
def fn1(n):
    result = 0
    for k in range(n):
        k+=1
        result += comb(n,k)*k**2
        
    return int(result)

fn2 = lambda n: int(2**(n-2)*(n)*(n+1))

In [46]:
for n in range(1,6):
    id1,id2 = fn1(n),fn2(n)
    if id1 == id2:
        print(f'for n={i}: combination 1: {id1} combination 2: {id2}')

for n=5: combination 1: 1 combination 2: 1
for n=5: combination 1: 6 combination 2: 6
for n=5: combination 1: 24 combination 2: 24
for n=5: combination 1: 80 combination 2: 80
for n=5: combination 1: 240 combination 2: 240


###### verification

a) for $n=1, k=1$

$
\begin{align}
 \sum_{k=1}^{n} {n \choose k} k^2 = 2^{n-2}n(n+1)
\end{align}
$

$
\begin{align}
  \implies {1 \choose 1} \cdot 1^2 = 2^{1-2}(1)(1+1)
\end{align}
$

$
\begin{align}
  \implies 1 = 1
\end{align}
$


b) for $n=2, k=1,2$

$
\begin{align}
  \implies {2 \choose 1} \cdot 1^2 + {2 \choose 2} \cdot 2^2 = 6
\end{align}
$

$
\begin{align}
  \implies 2^{2-2}(2)(2+1) = 6
\end{align}
$

$
\begin{align}
  \implies 6 = 6
\end{align}
$

c) for $n=3, k=1,2,3$

$
\begin{align}
  \implies {3 \choose 1} \cdot 1^2 + {3 \choose 2} \cdot 2^2 
   + {3 \choose 3} \cdot 3^2
   = 3^{3-2}(3)(3+1)
\end{align}
$

$
\begin{align}
  \implies 24 = 24
\end{align}
$

c) for $n=4, k=1,2,3,4$

$
\begin{align}
  \implies {4 \choose 1} \cdot 1^2 
   + {4 \choose 2} \cdot 2^2 
   + {4 \choose 3} \cdot 3^2
   + {4 \choose 4} \cdot 4^2
   = 4^{4-2}(4)(4+1)
\end{align}
$

$
\begin{align}
  \implies 80 = 80
\end{align}
$

c) for $n=5, k=1,2,3,4,5$

$
\begin{align}
  \implies {5 \choose 1} \cdot 1^2 
   + {5 \choose 2} \cdot 2^2 
   + {5 \choose 3} \cdot 3^2
   + {5 \choose 4} \cdot 4^2
   + {5 \choose 4} \cdot 5^2
   = 5^{5-2}(5)(5+1)
\end{align}
$

$
\begin{align}
  \implies 240 = 240
\end{align}
$

Combinatorial Proof:

k members (of the committee) can be selected from n people ${n \choose k}$ ways. One chairperson and one secretary can be selected from the committee (both can be the same $n \choose k$$k^2$ ways.

If the chair and the secretary are the same person then they can be $n 2^{n-1}$ because there are two possible states, in the group and out of the group and the exponent represents the total of the group less the chair and secretary. Finally the n value represents the total number of ways this can occur.

A chair person can be selected from a group of $n$ people $n \choose 1$ ways, and a secratary $n-1 \choose 1$ ways, which is the same as saying $n(n-1)$. 

Similar to above the remaining members can be selected $2^{n-2}$ ways. This results in $n(n-1)2^{n-2}$ for secretary and chair as different people.

Finally combining the two scenarios we get: $n 2^{n-1} + n(n-1)2^{n-2}$, which simplifies to $2^{n-2}(n+1)$
